bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/googletv/GoogleTVRequest.java

```diff
@@ -0,0 +1,151 @@
+            message = "080210c801522d0a19636f6d2e676f6f676c652e616e64726f69642e766964656f73121073616d73756e6720534d2d4739393855";
+        } else if (messageId == 2) {
+            // Unknown but required
+            // 080210c801a201 0e 0a 04 08031006 0a 04 08031004 1802
+            // ---------------LEN---LEN------------LEN
+            message = "080210c801a2010e0a04080310060a04080310041802";
+        } else if (messageId == 3) {
+            // Trigger PIN OSD
+            // ---------------LEN---LEN
+            // 080210c801a201 08 12 04 08031006 1801
+            // 080210c801f201 08 0a 04 08031006 1001
+            message = "080210c801f201080a04080310061001";
+        } else if (messageId == 4) {
+            // 0a41087e123d0a 08 534d2d4739393855 12 07 73616d73756e67 18 01 22 02 3133 2a
+            // ---------------LEN--SM-G998U----------LEN--samsung---------
+            // 19 636f6d2e676f6f676c652e616e64726f69642e766964656f73 32
+            // LEN-com.google.android.videos----------------------------
+            // 07 342e33382e3138
+            // LEN-4.38.18
+            // message =
+            // "0a41087e123d0a08534d2d4739393855120773616d73756e671801220231332a19636f6d2e676f6f676c652e616e64726f69642e766964656f733207342e33382e3138";
+
+            // 0a5708fe0412520a 08 534d2d4739393855 12 07 73616d73756e67 18 01 22 02 3133 2a
+            // -----------------LEN--SM-G998U----------LEN--samsung---------
+            // 19 636f6d2e676f6f676c652e616e64726f69642e766964656f73 32
+            // LEN-com.google.android.videos---------------------------
+            // 1c 342e33392e3538342e3532393538383538332e372d72656c65617365
+            // LEN-4.39.584.529588583.7-release
+            message = "0a5708fe0412520a08534d2d4739393855120773616d73756e671801220231332a19636f6d2e676f6f676c652e616e64726f69642e766964656f73321c342e33392e3538342e3532393538383538332e372d72656c65617365";
+        } else if (messageId == 5) {
+            // Unknown. Sent after "1200" received
+            message = "1202087e";
+        }
+        return message;
+    }
+
+    public static String keepAlive(String request) {
+        // 42 07 08 01 10 e4f1 8d01
+        // 4a 02 08 01
+
+        // 42 08 08 7f 10 b4 908a a819
+        // 4a 02 08 7f
+
+        // 42 09 08 8001 10 ed b78a a819
+        // 4a 03 08 8001
+
+        char[] charArray = request.toCharArray();
+        StringBuffer sb = new StringBuffer();
+        sb.append(request);
+        sb.setLength(sb.toString().length() - 6);
+        String st = "";
+        do {
+            int sbLen = sb.toString().length();
+            st = "" + charArray[sbLen - 2] + charArray[sbLen - 1];
+            if (!DELIMITER_10.equals(st)) {
+                sb.setLength(sbLen - 2);
+            }
+        } while (!DELIMITER_10.equals(st));
+        sb.setLength(sb.toString().length() - 2);
+
+        StringBuffer sbReply = new StringBuffer();
```

Comment:
There's no need of the thread-safety overehead of a StringBuffer here:

```suggestion
        StringBuilder sbReply = new StringBuilder();
```